### ISSUE TITLE
Do not extend from doctrine repository

### ISSUE THREAD

**Issue by fesor** [Arky9782/chat#2]
https://github.com/Arky9782/chat/blob/45a9451f3a13ff704858bd9afc52ce0ecad4076d/src/AppBundle/Repository/MessageRepository.php#L11

Doctrine repositories are general purpose persistence utility. It's not good idea to couple your application to such low-level stuff. Instead create your own repositories with specialized interface which will expose only things that client-code needs and nothing more. 

Basically your goal is to hide doctrine as much as possible and split functionality into specialized interfaces (in term of public class API, you don't need interfaces as language entity if you don't have multiple implementations).

```
class MessageRepository {
    // inject entity manager
    public function add(Message $message): void 
    {
         $this->em->persist($message);
    }
}
```
